### PR TITLE
NPI and MedAPI, Dependency Updates

### PR DESCRIPTION
**Pull request by oehokie**
Requires meta, model, cms, bb, match, generate, record PR's to be accepted and published

### CODE REVIEW

File: client/app/scripts/directives/entry.js [hunk 1]
```diff
@@ -178,18 +178,51 @@ angular.module('phrPrototypeApp')
                         $("#comments" + entryIndex).removeClass("in");
                         $("#history" + entryIndex).removeClass("in");
                         $("#match" + entryIndex).removeClass("in");
+                        $("#images" + entryIndex).removeClass("in");
+                        $("#fda" + entryIndex).removeClass("in");
+                        $("#medline" + entryIndex).removeClass("in");
```
Comment (jsachs) [inline client/app/scripts/directives/entry.js]: What's going on with this crazy branching? Is there a way to cut down on this? I may go in and write something in lodash to simplify.

File: lib/medAPI/index.js [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+            console.log("error: " + err);
+            res.status(400).send(err);
+        } else {
+            res.set('Content-Type', 'application/json');
+            res.status(200).send(data);
+        }
+    });
+});
+
+app.post('/api/v1/openfdaname', login.checkAuth, function (req, res) {
```
Comment (jsachs) [inline lib/medAPI/index.js]: Do we want to break this out to `/openfda/name` and `/openfda/code`?

Comment (oehokie) [inline lib/medAPI/index.js]: @ekavaliM ^^

File: lib/npiAPI/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        if (err2) {
+            console.log("Error: " + JSON.stringify(err2, null, 4));
+            callback(err2);
+        } else {
+            callback(null, results);
+        }
+    })
+}
+
+app.post('/api/v1/getnpi', function (req, res) {
```
Comment (jsachs) [inline lib/npiAPI/index.js]: This seems confusing. It's a `POST`, but the API route has 'get' in it.

File: server.js [hunk 4]
```diff
@@ -156,6 +156,12 @@ app.use(accountHistory);
 var notes = require('./lib/notes');
 app.use(notes);
 
+var medapi = require('./lib/medAPI');
```
Comment (jsachs) [inline server.js]: We might need to rename this when I merge the CRUD API for medications in.

File: lib/npiAPI/index.js (hunk 3 above)
Comment (ekavaliM) [inline lib/npiAPI/index.js]: API route was previously a GET and we switched over. We can modify it.

Comment (oehokie) [inline lib/npiAPI/index.js]: I fixed this, will push when i finish local installs/tests

File: lib/medAPI/index.js (hunk 2 above)
Comment (oehokie) [inline lib/medAPI/index.js]: i broke this out

**Pull request merged**